front-end/src/components/main-page/MainPage.jsx

```diff
@@ -29,14 +45,23 @@ class MainPage extends Component {
     }
   }
 
-  getAnalyzedData = async() =>{
+  getAnalyzedData = async() => {
       const override = store.getState().analyzeTranscript.override;
       const projectId = store.getState().analyzeTranscript.projectIdToBeDisplayed;
-      const transcripts = store.getState().analyzeTranscript.analyzedTranscripts;
-      if(override || (!(projectId in transcripts))){
+      const analyzedTranscripts = store.getState().analyzeTranscript.analyzedTranscripts;
+      console.log(projectId)
+      if(override || (!(projectId in analyzedTranscripts))){
+        // Call API if either the user wants to override or the project ID is not stored in this session's transcripts
         const analyzedData = await TranscriptAPI.getAnalysis({project_id: projectId});
-        store.dispatch(addAnalyzedTranscript({projectId: projectId, transcript: analyzedData.data}));
-        store.dispatch(setOverrideStatus(false));
+        if (analyzedData.data.length === 0) {
+          // Throw an error if getAnalysis returns an empty array, as this means no transcript with that project ID was found
+          const missingIdError = new Error();
+          missingIdError.response = {data: {error: "Your project ID was not in our database. Please try again."}}
```

Comment:
small grammar thing, the English in me is coming out, but might wanna do is-present-tense rather than was.

Reply:
Good catch!